src/api/v4/error.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Build a cartoError from a generic error
+ * @constructor
+ * 
+ * @return {CartoError} A well formed object representing the error.
+ */
+function CartoError (error) {
+  if (!error) {
```

Comment:
I think we can get rid of this if (we're also assigning 'unexpected error' to `this.message` in line #20)